src/libdrakvuf/libdrakvuf.h

```diff
@@ -274,6 +302,32 @@ char *drakvuf_get_current_process_name(drakvuf_t drakvuf,
                                        uint64_t vcpu_id,
                                        x86_registers_t *regs);
 
+
+bool drakvuf_get_current_thread_id( drakvuf_t drakvuf, 
+                                    uint64_t vcpu_id, 
+                                    x86_registers_t *regs,
+                                    uint32_t *thread_id );
+
+bool drakvuf_get_previous_mode( drakvuf_t drakvuf, 
```

Comment:
Yeap, that works. There should also be some validation that the value that was read matches the expected enum values instead of just blindly returning it to the user. If it's not one of the expected values the function should return false.